addons/udes_stock/tests/test_move.py

```diff
@@ -100,3 +100,31 @@ def test03_cancel_backorder_cancels_remaining(self):
 
         # Remaining move2 should be cancelled.
         self.assertEqual(new_move2.state, "cancel")
+
+
+class TestUnreserveIntialDemand(common.BaseUDES):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestUnreserveIntialDemand, cls).setUpClass()
+        cls.pack_4apples_info = [{'product': cls.apple, 'qty': 4}]
+        # Set target storage format to easily pick
+        cls.picking_type_pick.u_target_storage_format = 'product'
+
+    def test01_do_not_bypass_reservation_update_at_unreserve_initial_demand(self):
+        """ When calling _unreserve_initial_demand() disable bypass reservation update
+            which is now enabled by default.
+        """
+        Picking = self.env['stock.picking']
+        apple_quant = self.create_quant(self.apple.id, self.test_location_01.id, 4)
+        picking = self.create_picking(self.picking_type_pick, products_info=self.pack_4apples_info, confirm=True, assign=True)
+
+        self.assertEqual(picking.state, 'assigned')
+        self.assertEqual(apple_quant.reserved_quantity, 4)
+
+        partial_pick_apples = [{'barcode': self.apple.barcode, 'qty': 2}]
+
+        picking.update_picking(product_ids=partial_pick_apples, validate=True, create_backorder=True)
+        self.assertEqual(picking.state, 'done')
+        backorder = Picking.get_pickings(backorder_id=picking.id)
+        self.assertEqual(backorder.state, 'assigned')
```

Comment:
Is it worth asserting the remaining reserved quantity on the backorder picking too?

Reply:
The problem was it wasn't getting assigned at all so if the state is assigned means we have found stock, so the quantity is not really needed.